src/appleseed/renderer/modeling/surfaceshader/nprsurfaceshaderhelper.cpp

```diff
@@ -0,0 +1,248 @@
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2018 Esteban Tovagliari, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Interface header.
+#include "nprsurfaceshaderhelper.h"
+
+// appleseed.renderer headers.
+#include "renderer/kernel/aov/aovcomponents.h"
+#include "renderer/kernel/intersection/intersector.h"
+#include "renderer/kernel/npr/nprclosures.h"
+#include "renderer/kernel/shading/closures.h"
+#include "renderer/kernel/shading/shadingcomponents.h"
+#include "renderer/kernel/shading/shadingcontext.h"
+#include "renderer/kernel/shading/shadingpoint.h"
+#include "renderer/modeling/material/material.h"
+#include "renderer/modeling/shadergroup/shadergroup.h"
+
+// appleseed.foundation headers.
+#include "foundation/math/sampling/mappings.h"
+#include "foundation/utility/arena.h"
+
+// Standard headers.
+#include <cstddef>
+
+using namespace foundation;
+using namespace std;
+
+namespace renderer
+{
+
+void NPRSurfaceShaderHelper::evaluate(
+    SamplingContext&            sampling_context,
+    const ShadingContext&       shading_context,
+    const ShadingPoint&         shading_point,
+    AOVComponents&              components,
+    ShadingComponents&          radiance) const
+{
+    const Material* material = shading_point.get_material();
+    const ShaderGroup* sg = material->get_render_data().m_shader_group;
+
+    evaluate_npr(sampling_context, shading_context, shading_point, sg, radiance, components);
+}
+
+void NPRSurfaceShaderHelper::evaluate_npr(
+    SamplingContext&            sampling_context,
+    const ShadingContext&       shading_context,
+    const ShadingPoint&         shading_point,
+    const ShaderGroup*          sg,
+    ShadingComponents&          radiance,
+    AOVComponents&              components) const
+{
+    // For now, we only work in RGB mode.
+    assert(radiance.m_beauty.get_mode() == Spectrum::RGB);
```

Comment:
Maybe simply do nothing if not in RGB mode? That will avoid crashing or producing weird results.